test/runnable/interface2.d

```diff
@@ -920,6 +923,57 @@ void test27()
 }
 
 /*******************************************************/
+// 1747 & 2013
+
+void test1747()
+{
+    interface IA          { int mA(); }
+    interface IB : IA     { int mB(); }
+    interface IC : IB     { }
+    interface ID : IA, IC { int mD(); }
+
+    // offset:   0   +n  +n + ptrsize
+    //                  (IA)
+    //                   IB
+    //               IA, IC
+    static class C : ID
+    {
+        int mA() { return 1; }
+        int mB() { return 2; }
+        int mD() { return 3; }
+    }
+
+    C c = new C;    void* pc  = *cast(void**)&c;
+    ID id = c;      void* pid = *cast(void**)&id;
+    IC ic = c;      void* pic = *cast(void**)&ic;
+    IB ib = c;      void* pib = *cast(void**)&ib;
+    IA ia = c;      void* pia = *cast(void**)&ia;
```

Comment:
Such the casts implicitly tested below. These method calls through interface references:

```d
    assert(id.mA() == 1);
    assert(id.mB() == 2);   // OK <- NG (bugzilla 2013 case)
    assert(id.mD() == 3);

    assert(ic.mA() == 1);
    assert(ic.mB() == 2);   // OK <- NG (bugzilla 2013 case)

    assert(ib.mA() == 1);
    assert(ib.mB() == 2);   // OK <- NG

    assert(ia.mA() == 1);
```

Are equivalent with:

```d
    assert((cast(IA)id).mA() == 1);
    assert((cast(IB)id).mB() == 2);   // OK <- NG (bugzilla 2013 case)
    assert(         id .mD() == 3);

    assert((cast(IA)ic).mA() == 1);
    assert((cast(IB)ic).mB() == 2);   // OK <- NG (bugzilla 2013 case)

    assert((cast(IA)ib).mA() == 1);
    assert(         ib .mB() == 2);   // OK <- NG

    assert(         ia .mA() == 1);
```